File: 100-same-tree/100-same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        
        def tre(node):
            if node:
                return [node.val] + tre(node.left) + tre(node.right)
            else:
                return ['null']
        
        return tre(p) == tre(q)
            
            

File: 100-same-tree/100-same-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function(p, q) {
    
    const preorder_t = function(node){
        
        if(node != null){
         return [node.val].concat(preorder_t(node.left),preorder_t(node.right))   
        }
        else{
            return ['null']
        };
        
    };
    
    return JSON.stringify(preorder_t(p)) === JSON.stringify(preorder_t(q));
    
};